### PR TITLE
RXI-444 logs rotation

### PR DESCRIPTION
**Pull request by oleks-rip**

### CODE REVIEW

File: src/xbwd/app/App.cpp [hunk 1]
```diff
@@ -186,4 +194,81 @@ App::federator()
...
+                ripple::jv("msg", ec.message()));
+            continue;
+        }
+
+        if (size < logSize)
+            continue;
+
+        unsigned num = config_->logFilesToKeep;
+        std::filesystem::path const bak = makeBakPath(logFile, num);
+        std::filesystem::remove(bak, ec);
```
Comment (pwang200) [inline src/xbwd/app/App.cpp:260]: What happens when the file is removed while it is being written to?

File: src/xbwd/app/App.cpp [hunk 2]
```diff
@@ -186,4 +194,81 @@ App::federator()
...
+
+        for (; num > 0; --num)
+        {
+            std::filesystem::path const bakDst = makeBakPath(logFile, num);
+            std::filesystem::path const bakSrc = makeBakPath(logFile, num - 1);
+            std::filesystem::rename(bakSrc, bakDst, ec);
```
Comment (pwang200) [inline src/xbwd/app/App.cpp:266]: Similar to the comment above, what happens when the file is renamed while it is being written to?

File: src/xbwd/app/App.cpp (hunk 1 above)
Comment (oleks-rip) [reply src/xbwd/app/App.cpp:260]: std::filesystem::remove  just remove file name from directory, not the file itself. The file will be removed after there are no more any open handles on it.

File: src/xbwd/app/App.cpp (hunk 2 above)
Comment (oleks-rip) [reply src/xbwd/app/App.cpp:266]: It is the same - posix's file name has no connection to the open file handle. You can do whatever you want with the name.

**Review by pwang200 — APPROVED** [review]
LGTM

**Comment by seelabs** [comment]
merged

**Pull request closed without merging**